tests/always/test_example_dags.py

```diff
@@ -29,33 +30,38 @@
 NO_DB_QUERY_EXCEPTION = ["/airflow/example_dags/example_subdag_operator.py"]
 
 
-class TestExampleDags(unittest.TestCase):
-    def test_should_be_importable(self):
-        example_dags = list(glob(f"{ROOT_FOLDER}/airflow/**/example_dags/example_*.py", recursive=True))
-        assert 0 != len(example_dags)
-        for filepath in example_dags:
-            relative_filepath = os.path.relpath(filepath, ROOT_FOLDER)
-            with self.subTest(f"File {relative_filepath} should contain dags"):
-                dagbag = DagBag(
-                    dag_folder=filepath,
-                    include_examples=False,
-                )
-                assert 0 == len(dagbag.import_errors), f"import_errors={str(dagbag.import_errors)}"
-                assert len(dagbag.dag_ids) >= 1
-
-    def test_should_not_do_database_queries(self):
-        example_dags = glob(f"{ROOT_FOLDER}/airflow/**/example_dags/example_*.py", recursive=True)
-        example_dags = [
-            dag_file
-            for dag_file in example_dags
-            if any(not dag_file.endswith(e) for e in NO_DB_QUERY_EXCEPTION)
-        ]
-        assert 0 != len(example_dags)
-        for filepath in example_dags:
-            relative_filepath = os.path.relpath(filepath, ROOT_FOLDER)
-            with self.subTest(f"File {relative_filepath} shouldn't do database queries"):
-                with assert_queries_count(0):
-                    DagBag(
-                        dag_folder=filepath,
-                        include_examples=False,
-                    )
+def example_dags():
+    example_dirs = ["airflow/**/example_dags/example_*.py", "tests/system/providers/**/example_*.py"]
+    for example_dir in example_dirs:
+        yield from glob(f"{ROOT_FOLDER}/{example_dir}", recursive=True)
+
+
+def example_dags_except_db_exception():
+    return [
+        dag_file
+        for dag_file in example_dags()
+        if any(not dag_file.endswith(e) for e in NO_DB_QUERY_EXCEPTION)
+    ]
+
+
+def relative_path(path):
+    return os.path.relpath(path, ROOT_FOLDER)
+
+
+@pytest.mark.parametrize("example", list(example_dags()), ids=relative_path)
+def test_should_be_importable(example):
+    dagbag = DagBag(
+        dag_folder=example,
+        include_examples=False,
+    )
+    assert 0 == len(dagbag.import_errors), f"import_errors={str(dagbag.import_errors)}"
```

Comment:
Nit: I am not sure about this, but I believe the general guide is to use expected value on the right side of assert statement.
It is not different at all from the test status perspective, but may be more clear in the error message in case assert fails that `len(dagbag.import_errors)` actually has to be qual to zero.

Reply:
Nope. Pytest (thank you Pytest maintainers!) reverted the expected/actual vs. assertsEqual (which follows JUnit's practice of expected first):  https://stackoverflow.com/questions/53651193/what-is-the-correct-order-for-actual-and-expected-in-pytest

Reply:
The code is updated